src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,37 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final Random RANDOM = new Random();
+    private static final String COLOR = new ColorSupplier().getRandomColor();

```

Comment:
The COLOR constant should not be initialized here with a value from ColorSupplier. Instead, it should be generated within the getRandomFigure method to ensure a new color is provided for each figure.